File: Core/DotNet/MediaHatchCore/Movie.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MediaHatchCore
{
    public class Movie : Media
    {
        [JsonProperty("releaseYear")]
        public int ReleaseYear { get; set; }

        public override string FullName(string name = null)
        {
            if (ReleaseYear > 0)
                return $"{name ?? Name} ({ReleaseYear})";
            else
                return name ?? Name;
        }
    }
}

File: Core/media-hatch-core/src/movie.ts
import { Media } from './media';
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject()
export class Movie extends Media {
	public static typeName = 'MediaHatchCore.Movie';

	constructor() {
		super(Movie.typeName);
	}

	@JsonProperty('releaseYear')
	public releaseYear = 0;

	public get fullName(): string {
		if (this.releaseYear > 0) {
			return `${this.name} (${this.releaseYear})`;
		} else {
			return this.name;
		}
	}
}
